x-pack/plugins/observability/public/pages/slos/components/slo_list_group_by.tsx

```diff
@@ -32,104 +25,110 @@ export type Item<T> = EuiSelectableOption & {
-    }))
-  );
-
-  useEffect(() => {
-    setGroupByOptions(
-      GROUP_BY_OPTIONS.map((option) => ({
-        ...option,
-        checked: option.type === initialState ? 'on' : undefined,
-      }))
-    );
-  }, [initialState]);
-
-  const selectedGroupBy = groupByOptions.find((option) => option.checked === 'on');
-  const handleToggleGroupByButton = () => setGroupByPopoverOpen(!isGroupByPopoverOpen);
-
-  const handleChangeGroupBy = (newOptions: Array<Item<GroupByField>>) => {
-    setGroupByOptions(newOptions);
-    setGroupByPopoverOpen(false);
+export function SloGroupBy({ onStateChange, groupBy }: Props) {
+  const [isGroupByPopoverOpen, setIsGroupByPopoverOpen] = useState(false);
+  const handleChangeGroupBy = ({ value, label }: { value: GroupByField; label: string }) => {
+    setGroupLabel(label);
     onStateChange({
       page: 0,
-      groupBy: newOptions.find((o) => o.checked)!.type,
+      groupBy: value,
     });
   };
+  const groupByOptions: Option[] = [
+    {
+      label: i18n.translate('xpack.observability.slo.list.groupBy.sliIndicator', {
+        defaultMessage: 'None',
+      }),
+      checked: groupBy === 'ungrouped',
+      value: 'ungrouped',
+      onClick: () => {
+        handleChangeGroupBy({
+          value: 'ungrouped',
+          label: i18n.translate('xpack.observability.slo.list.groupBy.upgrouped', {
+            defaultMessage: 'None',
+          }),
+        });
+      },
+    },
+    {
+      label: i18n.translate('xpack.observability.slo.list.groupBy.tags', {
+        defaultMessage: 'Tags',
+      }),
+      checked: groupBy === 'slo.tags',
+      value: 'slo.tags',
+      onClick: () => {
+        handleChangeGroupBy({
+          value: 'slo.tags',
+          label: i18n.translate('xpack.observability.slo.list.groupBy.tags', {
+            defaultMessage: 'Tags',
+          }),
+        });
+      },
+    },
+    {
+      label: i18n.translate('xpack.observability.slo.list.groupBy.status', {
+        defaultMessage: 'Status',
+      }),
+      checked: groupBy === 'status',
+      value: 'status',
+      onClick: () => {
+        handleChangeGroupBy({
+          value: 'status',
+          label: i18n.translate('xpack.observability.slo.list.groupBy.status', {
+            defaultMessage: 'Status',
+          }),
+        });
+      },
+    },
+    {
+      label: i18n.translate('xpack.observability.slo.list.groupBy.sliType', {
+        defaultMessage: 'SLI type',
+      }),
+      checked: groupBy === 'slo.indicator.type',
+      value: 'slo.indicator.type',
+      onClick: () => {
+        handleChangeGroupBy({
+          value: 'slo.indicator.type',
+          label: i18n.translate('xpack.observability.slo.list.groupBy.sliType', {
+            defaultMessage: 'SLI type',
+          }),
+        });
+      },
+    },
+  ];
+
+  const [groupLabel, setGroupLabel] = useState(
+    groupByOptions.find((option) => option.value === groupBy)?.label || 'None'
```

Comment:
I think this is not working as expected:
When I select groupBy "status" and reload the page, the `status` is correctly applied except for the groupBy select menu, which shows "None".
I believe you need to use a useEffect() on the groupBy prop 🤔 ?

![image](https://github.com/elastic/kibana/assets/1376800/ae2532d3-0d2f-4835-8a84-3913689ce649)

![image](https://github.com/elastic/kibana/assets/1376800/0bcdefd9-1cc7-4196-aef2-6775861e433c)

Reply:
Oh yes! This reminds me of the issue we fixed with a `useEffect` https://github.com/elastic/kibana/pull/175436/commits/8f6b7d6026a08df88aa77028cf8365bead7a3261#diff-6abbde604d870c8b4206a92d5ace80bbf86aaae4c0b1208bdb3e867da2dd49a9R70

Reply:
Great catch! Thank you!